src/anomalib/models/image/openai_vlm/lightning_model.py

```diff
@@ -0,0 +1,325 @@
+
+        Returns:
+            str: The response from the GPT-4 model indicating whether the image has anomalies or not.
+                  It returns 'NO' if there are no anomalies and 'YES: description' if there are anomalies,
+                  where 'description' provides details of the anomaly and its position.
+
+        Raises:
+            openai.error.OpenAIError: If there is an error during the API call.
+        """
+
+        # Function to encode the image
+        def encode_image(image_path: str) -> str:
+            path = Path(image_path)
+            with path.open("rb") as image_file:
+                return base64.b64encode(image_file.read()).decode("utf-8")
+
+        # Getting the base64 string
+        base64_image = encode_image(image)
+        base64_image_pre = [encode_image(img) for img in pre_img]
+        prompt = """
+         You will receive an image that is going to be an example of the typical image without any anomaly,
+         and the last image that you need to decide if it has an anomaly or not.
+         Answer with a 'NO' if it does not have any anomalies and 'YES: description'
+         where description is a description of the anomaly provided, position.
+        """
+        messages = [
+            {
+                "role": "system",
+                "content": prompt,
+            },
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "image_url",
+                        "image_url": {"url": f"data:image/png;base64,{base64_image_pre[0]}"},
+                        "detail": "high",
+                    },
+                ],
+            },
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "image_url",
+                        "image_url": {"url": f"data:image/png;base64,{base64_image}"},
+                        "detail": "high",
+                    },
+                ],
+            },
+        ]
+
+        try:
+            # Make the API call using the openai library
+            response = self.openai.chat.completions.create(
+                model="gpt-4o-mini-2024-07-18",
+                messages=messages,
+                max_tokens=300,
+            )
+            return response.choices[-1].message.content
+        except openai.APIConnectionError as e:
+            print("The server could not be reached")
+            print(e.__cause__)  # an underlying Exception, likely raised within httpx.
+            raise
+        except openai.RateLimitError as e:
+            print("A 429 status code was received; we should back off a bit.")
+            print(e.__cause__)
+            raise
+        except openai.APIStatusError as e:
+            print("Another non-200-range status code was received")
+            print(e.status_code)
+            print(e.response)
+            raise
+
+    def api_call(self, image: str) -> str:
```

Comment:
There seems to be some duplication between this method and `api_call_few_shot` which could be removed by refactoring.